File: programas/tarea9.java
/*
Tarea 9
Convertidor de monedas
Estudiante: Erik Everardo Cavazos Hernández
Matrícula: 1811290
Clase: Lenguajes de Programación V4-V6 Jueves
Docente: Ismael Pimentel
*/
import javax.swing.JOptionPane;

public class tarea9{
    public static void main(String[] args){
        //definicion de constantes (1 peso en...)
        final float dolar = 0.050f;
        final float euro = 0.044f;
        final float yen = 5.65f;
        final float dolarHK = 0.39f;

        //strings de botones
        String[] opciones = {"Dolar estadounidense", "Euro", "Yen (japones)", "Dolar HK"};
        String[] opcionesSalir = {"Si", "No"};
        float resultado;
        int salir;

        do{
            float entrada = Float.parseFloat(JOptionPane.showInputDialog(null,"Introducir valor en pesos mexicanos"));
            int accion = JOptionPane.showOptionDialog(
                null, 
                "Convertir " + entrada + " pesos a...", 
                "Convertir a", 
                JOptionPane.YES_NO_CANCEL_OPTION, 
                JOptionPane.INFORMATION_MESSAGE, 
                null, 
                opciones, 
                opciones[0]
                );
            switch(accion){
                case 0: 
                    resultado = entrada * dolar;
                    JOptionPane.showMessageDialog(null,entrada + " pesos equivale a " + resultado + " dolares");
                    historial.add(entrada + " pesos equivale a " + resultado + " dolares");
                    break;
                case 1: 
                    resultado = entrada * euro;
                    JOptionPane.showMessageDialog(null,entrada + " pesos equivale a " + resultado + " euros");
                    historial.add(entrada + " pesos equivale a " + resultado + " dolares");
                    break;
                case 2: 
                    resultado = entrada * yen;
                    JOptionPane.showMessageDialog(null,entrada + " pesos equivale a " + resultado + " yenes japonenes");
                    historial.add(entrada + " pesos equivale a " + resultado + " dolares");
                    break;
                case 3: 
                    resultado = entrada * dolarHK;
                    JOptionPane.showMessageDialog(null,entrada + " pesos equivale a " + resultado + " dolares de Hong Kong");
                    historial.add(entrada + " pesos equivale a " + resultado + " dolares");
                    break;
            }
            salir = JOptionPane.showOptionDialog(null,"Desea realizar otra conversion?", "Realizar otra operacion", JOptionPane.YES_NO_CANCEL_OPTION,JOptionPane.INFORMATION_MESSAGE,null, opcionesSalir,opcionesSalir[0]);
        }while(salir == 0);
    }
}

File: programas/tarea9.rb
#Tarea 9
#Convertidor de divisas
#Nombre: Erik Everardo Cavazos Hernandez
#Matrícula: 1811290
#Clase: Lenguajes de programación V4-V6 Jueves
#Docente: Ismael Pimentel
DOLAR = 0.050
EURO = 0.044
YEN = 5.65
DOLARHK = 0.39

begin
    puts "Convertir pesos a:"
    puts "1. Dolar"
    puts "2. Euro"
    puts "3. Yen Japones"
    puts "4. Dolar Hong Kong"
    print "=>"
    accion = gets().to_i
    print "Convertir: "
    entrada = gets().to_f
    case accion
    when 1 then 
        resultado = entrada * DOLAR
        puts "#{entrada} pesos es #{resultado} dolares"
    when 2 then 
        resultado = entrada * EURO
        puts "#{entrada} pesos es #{resultado} euros"
    when 3 then 
        resultado = entrada * YEN
        puts "#{entrada} pesos es #{resultado} yenes"
    when 4 then 
        resultado = entrada * DOLARHK
        puts "#{entrada} pesos es #{resultado} dolares de Hong Kong"
    else puts "Error"
    end
    print "Desea realizar otra conversión? (1. si 2. no): "
    si_no = gets()
end while si_no == 1